internal/storage/payload.go

```diff
@@ -26,18 +26,18 @@ import (
 
 // PayloadWriterInterface abstracts PayloadWriter
 type PayloadWriterInterface interface {
-	AddDataToPayload(msgs any, dim ...int) error
-	AddBoolToPayload(msgs []bool) error
-	AddByteToPayload(msgs []byte) error
-	AddInt8ToPayload(msgs []int8) error
-	AddInt16ToPayload(msgs []int16) error
-	AddInt32ToPayload(msgs []int32) error
-	AddInt64ToPayload(msgs []int64) error
-	AddFloatToPayload(msgs []float32) error
-	AddDoubleToPayload(msgs []float64) error
-	AddOneStringToPayload(msgs string) error
-	AddOneArrayToPayload(msg *schemapb.ScalarField) error
-	AddOneJSONToPayload(msg []byte) error
+	AddDataToPayload(msgs any, valids []bool) error
+	AddBoolToPayload(msgs []bool, valids []bool) error
+	AddByteToPayload(msgs []byte, valids []bool) error
+	AddInt8ToPayload(msgs []int8, valids []bool) error
+	AddInt16ToPayload(msgs []int16, valids []bool) error
+	AddInt32ToPayload(msgs []int32, valids []bool) error
+	AddInt64ToPayload(msgs []int64, valids []bool) error
+	AddFloatToPayload(msgs []float32, valids []bool) error
+	AddDoubleToPayload(msgs []float64, valids []bool) error
+	AddOneStringToPayload(msgs string, valids []bool) error
```

Comment:
Since there is only one element added, the `valids` slice looks confusing. It would be better if it looks like `AddOneStringToPayload(msgs string, valid bool) error`, or just determining null value by `msgs == nil` in the function implementation.

The same to all following `AddOneXXX` methods.